### PR TITLE
Fix warning about maxHeight in Popover

### LINKED ISSUE #16165: Unrecognized prop warning in
```
react_devtools_backend.js:2557 Warning: React does not recognize the `maxHeight` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `maxheight` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in span (created by TooltipPopover)
    in div
    in span
```

**To Reproduce**

1. Open orders table from a custom question
2. Click on the small grid icon to create Custom column
3. Focus on the "Formula" input, the topmost in the popup
4. Filter console by `Popover`

**Expected behavior**

No warning.

**Screenshots**

image

### PR DESCRIPTION
**Pull request by gusaiani**
Fixes #16165

Also corrects a few typos in TooltipPopover comments.

**How to test**

1. Open orders table from a custom question
2. Click on the small grid icon to create Custom column
3. Focus on the "Formula" input, the topmost in the popup
4. Filter console by `Popover`

You should see no warning.

### CODE REVIEW

**Review by daltojohnso — CHANGES_REQUESTED** [review]
`Popover` blindly passes `maxHeight` to its `children` because a number of its children in the app (`AccordionList`, `FieldList` to name a couple) have a `maxHeight` prop, so changing it to pass `"max-height"` fixes the warnings but theoretically breaks stuff elsewhere. All of the `maxHeight` code in `Popover` is pretty gross tbh, since we're obviously spraying this prop at things that don't need it, thus creating a bunch of react warnings.

We recently changed how popover height is calc'd when the `sizeToFit` prop is set. I don't know why we have both `sizeToFIt` and a `maxHeight`, but I'm not entirely sure that we need this `maxHeight` code at all, though I could be wrong. You could try removing it entirely and then we can test how popovers with large height values behave.

**Comment by gusaiani** [comment]
Thank you @daltojohnso 

I removed `maxHeight` but kept `childProps` declared as `{}`.

Just trying to avoid getting side-tracked here as the function using this const looks like

~~~js
        {typeof this.props.children === "function"
          ? this.props.children(childProps)
          : React.Children.count(this.props.children) === 1 &&
            // NOTE: workaround for https://github.com/facebook/react/issues/12136
            !Array.isArray(this.props.children)
          ? React.cloneElement(
              React.Children.only(this.props.children),
              childProps,
            )
          : this.props.children}
~~~

hopefully we can make sense of this at some point, or just use React.Portal or something.

**Comment by gusaiani** [comment]
@daltojohnso @nemanjaglumac, perhaps you may quickly know where the Popover can be checked when using the product.

Just to ensure the height is what we want it to be.

**Comment by daltojohnso** [comment]
I'll take a look tomorrow. I'm pretty sure we're using the maxHeight prop to prevent dropdowns with lots of options from expanding beyond the height of the browser... which is what setting `sizeToFit` also does. So I'll poke around and see if that's still the case. Worst case here I think we just need to turn `sizeToFit` on in a few situations.

File: frontend/src/metabase/components/Popover.jsx [hunk 1]
```diff
@@ -127,9 +127,7 @@ export default class Popover extends Component {
   }
 
   _popoverComponent() {
-    const childProps = {
-      maxHeight: this._getMaxHeight(),
-    };
+    const childProps = {};
```
Comment (kulyk) [inline frontend/src/metabase/components/Popover.jsx:130]: Should we also remove the `_getMaxHeight` method?

Comment (gusaiani) [reply frontend/src/metabase/components/Popover.jsx:128]: Removed @kulyk, let's see if the changes make it through.

Thank you 🙇

**Pull request closed without merging**